model/fv_control.F90

```diff
@@ -734,6 +733,27 @@ subroutine fv_control_init(Atm, dt_atmos, this_grid, grids_on_this_pe, p_split,
      Atm(this_grid)%neststruct%parent_proc = ANY(Atm(this_grid)%neststruct%child_grids) !ANY(tile_coarse == Atm(this_grid)%global_tile)
      Atm(this_grid)%neststruct%child_proc = ASSOCIATED(Atm(this_grid)%parent_grid) !this means a nested grid
 
+     if(n>1) then
+       call mpp_set_current_pelist( global_pelist )
+       do n=2,ngrids
+        ! Construct the MPI communicators that are used in fill_nested_grid_cpl()
+          allocate(Atm(n)%Bcast_ranks(1+size(Atm(n)%pelist)))
+          ! parent grid sending rank within Bcast_ranks array
+          Atm(n)%Bcast_ranks(1)=Atm(n)%parent_grid%pelist(1) + &
+                     ( Atm(n)%neststruct%parent_tile-tile_fine(Atm(n)%parent_grid%grid_number)+ &
+                       Atm(n)%parent_grid%flagstruct%ntiles-1 )*Atm(n)%parent_grid%npes_per_tile
+
```

Comment:
I think this logic could be simplified by using the entries in the Atm type.  Something like:

if (Atm(n)%neststruct%parent_tile == mpp_get_tile_id(Atm(n)%parent_grid%domain)) Atm(n)%Bcast_ranks(1) = mpp_get_domain_tile_root_pe(Atm(n)%parent_grid%domain)

Reply:
I think I yanked that logic from atmos_cubed_sphere/tools/fv_restart.F90 line 885.  I'll try a compile/run using your suggestion.

Reply:
atmos_cubed_sphere/model/fv_control.F90(742): error #6512: A scalar-valued expression is required in this context.
`if(Atm(n)%neststruct%parent_tile == mpp_get_tile_id(Atm(n)%parent_grid%domain)) then`
-------------------------------------^
I'll look at it more tomorrow.

Reply:
mpp_get_tile_id() returns an integer array.
line 1113 of mpp/include/mpp_domains_util.inc
`integer, dimension(size(domain%tile_id(:))) :: mpp_get_tile_id`
so I need to dump the output to the existing tile_id(1) variable.

my proposal becomes
`tile_id = mpp_get_tile_id(Atm(n)%domain)`
`if( Atm(n)%neststruct%parent_tile == tile_id(1) ) then`
`  Atm(n)%Bcast_ranks(1) = mpp_get_domain_tile_root_pe(Atm(n)%parent_grid%domain)`
`endif`

Reply:
Thanks for working through the issues with my napkin programming.  If that works and looks cleaner to you, then I'll approve the PR

Reply:
At least for the HAFS case I have, the above 'if' test is failing and results in the Bcast_ranks(1) entry not getting set.  My current code looks like this. 

`        ! parent grid sending rank within Bcast_ranks array`
`        tile_id = mpp_get_tile_id(Atm(n)%domain)`
`        if( Atm(n)%neststruct%parent_tile == tile_id(1) ) then`
`          write(6,'("fv_control_init: Setting lead rank for grid using new way ",2I4)') n,Atm(n)%neststruct%parent_tile`
`           Atm(n)%Bcast_ranks(1) = mpp_get_domain_tile_root_pe(Atm(n)%parent_grid%domain)`
`          else`
`           write(6,'("fv_control_init: Setting lead rank for grid old way ",2I4)') n,Atm(n)%neststruct%parent_tile`
`          Atm(n)%Bcast_ranks(1)=Atm(n)%parent_grid%pelist(1) + &`
`                   ( Atm(n)%neststruct%parent_tile-tile_fine(Atm(n)%parent_grid%grid_number)+ &`
`                      Atm(n)%parent_grid%flagstruct%ntiles-1 )*Atm(n)%parent_grid%npes_per_tile`
`          endif`

 I don't feel like debugging the new approach, so I vote keeping the original (ugly) approach.

Reply:
I think I see the issue with your modification, but don't think it's worth continuing to debug.  For documentation purposes, I think the if-test should be an ANY and not specifically against tile_id(1).

Reply:
Will test with any()

Reply:
Still going through the ugly path.

`if( any(Atm(n)%neststruct%parent_tile == tile_id) ) then`

Reply:
I think the tile_id should for the parent grid, i.e. _mpp_get_tile_id(Atm(n)%parent_grid%domain)_.  My example with the mismatched if-test was using that domain.

Reply:
`tile_id = mpp_get_tile_id(Atm(n)%parent_grid%domain)`

Does run, however, the parent grid ranks are still going through the old path while the nest ranks go through the new path.

Reply:
Can we move forward with the ugly code?